spec/counter_culture_spec.rb

```diff
@@ -2967,4 +2967,267 @@ def mess_up_counts
       expect { PostLike.create!(post: post) }.to change { post.reload.updated_at }
     end
   end
+
+  context "with aggregate_counter_updates" do
+    it "updates counter caches" do
+      user = User.create
+      product1 = Product.create
+      product2 = Product.create
+      product3 = Product.create
+
+      expect(user.reviews_count).to eq(0)
+      expect(product1.reviews_count).to eq(0)
+      expect(product2.reviews_count).to eq(0)
+      expect(product3.reviews_count).to eq(0)
+      expect(user.review_approvals_count).to eq(0)
+
+      review_to_delete = CounterCulture.aggregate_counter_updates do
+        user.reviews.create :user_id => user.id, :product_id => product1.id, :approvals => 5
+      end
+
+      user.reload
+      product1.reload
+
+      expect(user.reviews_count).to eq(1)
+      expect(user.using_count).to eq(1)
+      expect(user.review_approvals_count).to eq(5)
+      expect(user.dynamic_delta_count).to eq(1)
+      expect(user.custom_delta_count).to eq(3)
+      expect(product1.reviews_count).to eq(1)
+      expect(product1.rexiews_count).to eq(1)
+
+      CounterCulture.aggregate_counter_updates do
```

Comment:
How do you feel about using that new `expect_queries` helper in some of these tests?

Reply:
I wrote a bigger test for this